### PR TITLE
[17.0][MIG] account_payment_return_financial_risk: Migration to version 17.0

### PR DESCRIPTION
**Pull request by carlos-lopez-tecnativa**
@Tecnativa TT49282

Depends on:

- https://github.com/OCA/account-payment/pull/752

### CODE REVIEW

**Comment by carlosdauden** [comment]
Can you add account_payment_return repo dependency to tray in runboat?

**Comment by carlos-lopez-tecnativa** [comment]
> Can you add account_payment_return repo dependency to tray in runboat?

@carlosdauden done, could you please update review

**Review by carlosdauden — APPROVED** [review]

**Comment by carlos-lopez-tecnativa** [comment]
@carolinafernandez-tecnativa  @pedrobaeza could you please help me with the review of this module? Thank you!

File: account_payment_return_financial_risk/readme/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+To use this module, you need to:
+
+1.  Go to *Customers \> Financial Risk*
+2.  Set limits and choose options to compute in credit limit.
+3.  Create an invoice and pay it.
+4.  Create a payment return.
+5.  Go to *Invoicing/Accounting \> Customers \> Invoices* and create new
+    customer invoices.
+6.  Test the restriction trying to create an invoice for the partner for
+    an amount higher of the limit you have set.
```
Comment (carolinafernandez-tecnativa) [inline account_payment_return_financial_risk/readme/USAGE.md:10]: It is missing the steps you need to set up which field in financial risk.
Maybe is needed a CONFIGURATION.md, to explain we have to add groups to user Financial/Manager to allow to edit the amounts, and which field we need to set up.

Also here is missing the step to create a customer return

File: account_payment_return_financial_risk/readme/DESCRIPTION.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Extends Account Financial Risk to manage payments returns.
+
+If any limit is exceed the partner gets forbidden to confirm sale
```
Comment (carolinafernandez-tecnativa) [inline account_payment_return_financial_risk/readme/DESCRIPTION.md:3]: This does not belong to this module description
As i understand, you wont be able to confirm a payment return if the limit is exceed.

**Review by carolinafernandez-tecnativa — COMMENTED** [review]
I tested with this customer with this set up:

image

I created this invoice

image


I registered full payment and i go to create customer payment return for more than limit established

And i could confirm without the control

image

File: account_payment_return_financial_risk/readme/DESCRIPTION.md (hunk 2 above)
Comment (carlos-lopez-tecnativa) [reply account_payment_return_financial_risk/readme/DESCRIPTION.md:3]: @carolinafernandez-tecnativa  the purpose of this module is to include the amount of payment returns, but it is not restricted to validating a payment return. @carlosdauden, could you clarify the scope of the module?

@carolinafernandez-tecnativa  please remember that this module is just an additional extension of the main module https://github.com/OCA/credit-control/tree/17.0/account_financial_risk

image

Comment (carlos-lopez-tecnativa) [reply account_payment_return_financial_risk/readme/DESCRIPTION.md:3]: Ping @carlosdauden could you please give your feedback on the above question?

File: account_payment_return_financial_risk/readme/USAGE.md (hunk 1 above)
Comment (carlos-lopez-tecnativa) [reply account_payment_return_financial_risk/readme/USAGE.md:10]: It is in the original module, so I don't think we need to add the same in this module if it is only an extension.

https://github.com/OCA/credit-control/blob/17.0/account_financial_risk/readme/CONFIGURE.md

Comment (pedrobaeza) [reply account_payment_return_financial_risk/readme/USAGE.md:10]: You can add a mention saying "Refer to the module ... configuration for initial steps.".

File: account_payment_return_financial_risk/readme/DESCRIPTION.md (hunk 2 above)
Comment (carlosdauden) [reply account_payment_return_financial_risk/readme/DESCRIPTION.md:3]: Carlos you are right, what this module does is to add a new section to the financial risk to be able to manage that amount in the same way as the rest.

**Comment by carlosdauden** [comment]
Ready to merge

**Comment by carlos-lopez-tecnativa** [comment]
ping @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**